plugins/restful/RestfulFilesUpload.php

```diff
@@ -26,6 +26,8 @@ public static function controllersInfo() {
    * - "validators": By default no validation is done on the file extensions or
    *   file size.
    * - "scheme": By default the default scheme (e.g. public, private) is used.
+   * - "make_permanent": When true the file's status will be set to permanent
```

Comment:
It's been a while since I helped with the File API but I'm fairly certain they're more orthogonal. The usage is for reference counting to avoid deleting files in use. file_delete checks the references before deleting. So technically you could just skip setting the permanent status and add a usage, that would prevent it from being deleted but it would end up with a message in the logs every time cron ran. 

Adding the usage will prevent the file from being deleted if you're putting the fid into image or file fields, then remove it.